tests/cassandra_utils_test.py

```diff
@@ -407,6 +408,166 @@ def test_parsing_custom_seed_provider(self):
+            ssh=None,
+            checks=None,
+            logging=None,
+        )
+
+        c = Cassandra(medusa_config)
+
+        self.assertEqual(c.storage_port, 7000)
+        # Note: When not setting native_transport_port_ssl, still expecting that encrypted over existing
+        # native_transport_port.
+        self.assertEqual(c.native_port, 9042)
+        self.assertEqual(c.rpc_port, 9160)
+
+    def test_cassandra_client_encryption_enabled_ssl_port(self):
+        config = configparser.ConfigParser(interpolation=None)
+        config['cassandra'] = {
+            'config_file': os.path.join(os.path.dirname(__file__),
+                                        'resources/yaml/original/cassandra-client-encrypt-sslport.yaml'),
+            'start_cmd': '/etc/init.d/cassandra start',
+            'stop_cmd': '/etc/init.d/cassandra stop',
+            'is_ccm': '1'
+        }
+        config["grpc"] = {
+            "enabled": "0"
+        }
+        config['kubernetes'] = {
+            "enabled": "0"
+        }
+        medusa_config = MedusaConfig(
+            storage=None,
+            monitoring=None,
+            cassandra=_namedtuple_from_dict(CassandraConfig, config['cassandra']),
+            grpc=_namedtuple_from_dict(GrpcConfig, config['grpc']),
+            kubernetes=_namedtuple_from_dict(KubernetesConfig, config['kubernetes']),
+            ssh=None,
+            checks=None,
+            logging=None,
+        )
+
+        c = Cassandra(medusa_config)
+
+        self.assertEqual(c.storage_port, 7001)
+        # Note: When setting native_transport_port_ssl, expecting that non-encrypted over existing
+        # native_transport_port and encrypted over the native_transport_port_ssl
+        self.assertEqual(c.native_port, 9142)
+        self.assertEqual(c.rpc_port, 9160)
+
+    def test_cassandra_missing_native_port(self):
+        config = configparser.ConfigParser(interpolation=None)
+        config['cassandra'] = {
+            'config_file': os.path.join(os.path.dirname(__file__),
+                                        'resources/yaml/original/cassandra-missing-native-port.yaml'),
+            'start_cmd': '/etc/init.d/cassandra start',
+            'stop_cmd': '/etc/init.d/cassandra stop',
+            'is_ccm': '1'
+        }
+        config["grpc"] = {
+            "enabled": "0"
+        }
+        config['kubernetes'] = {
+            "enabled": "0"
+        }
+        medusa_config = MedusaConfig(
+            storage=None,
+            monitoring=None,
+            cassandra=_namedtuple_from_dict(CassandraConfig, config['cassandra']),
+            grpc=_namedtuple_from_dict(GrpcConfig, config['grpc']),
+            kubernetes=_namedtuple_from_dict(KubernetesConfig, config['kubernetes']),
+            ssh=None,
+            checks=None,
+            logging=None,
+        )
+
+        self.assertRaises(CassandraConfigInvalidError, Cassandra, medusa_config)
+
```

Comment:
Awesome sauce 💯 

We may have to drop the last test as valid configurations can avoid defining the ports and fallback to defaults.